src/linear_solver_routines/block_solver_helpers/gentlemans_house.jl

```diff
@@ -0,0 +1,155 @@
+
+# Constructor for Gent data structure
+function GentData(A::AbstractMatrix, bsize::Int64)
+    S = eltype(A)
+    m, n = size(A)
+    B = zeros(S, n + bsize + 1, n + 1)
+    @assert bsize >= 1 "bsize must be at least 1."
+    R = @views UpperTriangular(B[1:n, 1:n])
+    tab = @views B[1:n, n + 1]
+    v = zeros(n + 1) #zeros(n + bsize + 1)
+    return GentData{S, Matrix{S}, Vector{S}}(A, B, R, tab, v, bsize)
+end
+
+"""
+    gentleman!(G::GentData)
+
+A function that updates the `GenData` structure with the Gentleman's
+algorithm applied to a single row block of `A`.
+
+# Inputs
+- `G::GentData`, A data structure with the information to be updated
+by an iteration of Gentlemans.
+
+# Output
+The function will result in updates to the upper triangular part `R`
+and the `tab` vector of the `GentData` structure.
+"""
+function gentleman!(G::GentData)
+    B = G.B
+    v = G.v
+    m, n = size(B)
+    # Perform qr decomposition on B
+    LAPACK.geqrf!(B,v)
+    # Zero out the everything in the lower triangular part
+    for i in 1:n
+        for j in (i+1):m
+            @inbounds B[j,i] = 0
+        end
+
+    end
+
+    return nothing
+end
+
+#Function that zeros the gentleman's memory
+"""
+    resetGent!(G::GentData)
+
+Function that sets all the allocated memory of
+`GentData` to zero.
+"""
+function reset_gent!(G::GentData)
+    S = eltype(G)
+    fill!(G.B, zero(S))
+    fill!(G.v, zero(S))
+    return nothing
+end
+
+#Least squares solver for Gentleman's
+"""
+    ldiv!(x::AbstractVector, G::GentData, b::AbstractVector)
+
+Function that takes the allocated `GentData` structure to solve
+the least squares problem \$\\min_x \\|A x-b\\|_2^2\$. The function
+overwrites `x` with the solution.
+"""
+function LinearAlgebra.ldiv!(x::AbstractVector, G::GentData, b::AbstractVector)
+    m,n = size(G.A)
+    bsize = G.bsize
+    remb = rem(m, bsize)
+    # Detemine how many blocks by dividing by bsize and adding one if there is a remainder
+    nblocks = div(m, bsize) + (remb > 1 ? 1 : 0)
+    brows = n + bsize + 1
+    for i in 1:nblocks
+        # Do not use an index greater than m
+        index = (i - 1) * bsize + 1: min(i * bsize, m)
+        copy_block_from_mat!(G.B, G.A, b, index)
+        # Check if you are in the last block in which case zero all rows 
+        # that no data was moved to
+        if index[end] < i * bsize
+            fill!(view(G.B, (length(index) + 1) + n + 2:brows, :), zero(eltype(A)))
+        end
+
+        gentleman!(G)
+    end
+
+    # Solve Upper triangular system 
+    LinearAlgebra.ldiv!(x, G.R, G.tab)
+    reset_gent!(G)
+    return nothing 
+
+end
+
+# Function that copies the new block to the matrix being orthogonalized
+"""
+    copyBlockFromMat!(B::AbstractMatrix, A::AbstractMatrix, b::AbstractVector, index::Union{UnitRange{Int64}, Vector{Int64}})
+
+Function that updates matrix `B` with the entries at indicies `index` of the `A` matrix and `b` constant vector of the linear 
```

Comment:
Specify that the indices correspond to the rows of `A` and the entries of `b`

Reply:
I have added this specification